Blazor startup topic (#22279)
dotnet/AspNetCore.Docs: aspnetcore/blazor/fundamentals/environments.md

* [Blazor start configuration](#set-the-environment-via-startup-configuration)
* [`blazor-environment` header](#set-the-environment-via-header)

::: moniker-end

The client-side Blazor app (**`Client`**) of a hosted Blazor WebAssembly solution determines the environment from the **`Server`** app of the solution via a middleware that communicates the environment to the browser. The **`Server`** app adds a header named `blazor-environment` with the environment as the value of the header. The **`Client`** app reads the header. The **`Server`** app of the solution is an ASP.NET Core app, so more information on how to configure the environment is found in <xref:fundamentals/environments>.

For a standalone Blazor WebAssembly app running locally, the development server adds the `blazor-environment` header to specify the Development environment.

::: moniker range=">= aspnetcore-5.0"

## Set the environment via startup configuration

The following example starts Blazor in the Staging environment:

```cshtml
<body>
    ...

    <script src="_framework/blazor.{webassembly|server}.js" autostart="false"></script>
    <script>
      Blazor.start({

>         environment: Staging

Reviewer: ```suggestion
        environment: "Staging"
```

      });
    </script>
</body>
```

> Setting the environment using this approach overrides setting the environment with the [`blazor-environment` header](#set-the-environment-via-header).

Reviewer: ```suggestion
Setting the environment using this approach overrides setting the provided when the environment is set via the [`blazor-environment` header](#set-the-environment-via-header).
```

I think this is clearer?

Author: I'll try again ...

```suggestion
Using the `environment` property overrides the environment set by the [`blazor-environment` header](#set-the-environment-via-header).
```
